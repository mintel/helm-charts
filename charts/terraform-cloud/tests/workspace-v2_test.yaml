suite: Test Workspaces V2
templates:
  - workspace.yaml
release:
  namespace: test-namespace
tests:
  - it: Test simple workspace
    set:
      global.name: mntl-test-workspace
      global.clusterEnv: dev
      global.clusterName: cluster1
      global.clusterRegion: eu-west-1
      global.owner: sre
      global.partOf: test-project
      global.terraform.operatorVersion: v2
      global.terraform.agentPoolID: test-agent-pool
      s3:
        enabled: true
    asserts:
      - matchSnapshot: {} # Check for regressions and unexpected changes.
      - equal:
          path: metadata.labels.name
          value: mntl-test-workspace-s3
        documentIndex: 0
      - equal:
          path: spec.name
          value: dev-eu-west-1-cluster1-test-namespace-mntl-test-workspace-s3
        documentIndex: 0
      - equal:
          path: spec.organization
          value: Mintel
        documentIndex: 0
      - equal:
          path: spec.executionMode
          value: agent
        documentIndex: 0
      - equal:
          path: spec.applyMethod
          value: auto 
        documentIndex: 0
      - equal:
          path: spec.allowDestroyPlan
          value: true
        documentIndex: 0
      - equal:
          path: spec.agentPool.id
          value: test-agent-pool
        documentIndex: 0
      - equal:
          path: spec.sshKey.name
          value: mintel-ssh
        documentIndex: 0
      - equal:
          path: spec.token.secretKeyRef.key
          value: token
        documentIndex: 0
      - equal:
          path: spec.token.secretKeyRef.name
          value: terraformrc
        documentIndex: 0
