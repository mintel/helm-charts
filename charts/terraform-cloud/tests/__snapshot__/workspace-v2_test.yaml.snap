Test simple workspace:
  1: |
    apiVersion: app.terraform.io/v1alpha2
    kind: Workspace
    metadata:
      annotations:
        app.mintel.com/altManifestFileSuffix: mntl-test-workspace-s3
        app.mintel.com/placeholder: placeholder
        app.mintel.com/terraform-allow-destroy: "true"
        app.mintel.com/terraform-cloud-tags: env:dev,owner:sre,mod:s3
        app.mintel.com/terraform-owner: sre
        argocd.argoproj.io/sync-options: SkipDryRunOnMissingResource=true
        argocd.argoproj.io/sync-wave: "-40"
      labels:
        app.kubernetes.io/name: mntl-test-workspace-s3
        app.mintel.com/env: dev
        app.mintel.com/owner: sre
        app.mintel.com/region: eu-west-1
        name: mntl-test-workspace-s3
      name: dev-eu-west-1-cluster1-test-namespace-mntl-test-workspace-s3
      namespace: test-namespace
    spec:
      agentPool:
        id: test-agent-pool
      allowDestroyPlan: true
      applyMethod: auto
      executionMode: agent
      name: dev-eu-west-1-cluster1-test-namespace-mntl-test-workspace-s3
      organization: Mintel
      sshKey:
        name: mintel-ssh
      terraformVariables:
        - hcl: false
          name: aws_account_name
          sensitive: false
          value: dev
        - hcl: false
          name: aws_region
          sensitive: false
          value: eu-west-1
        - hcl: false
          name: eks_cluster_name
          sensitive: false
          value: cluster1
        - hcl: false
          name: enable_versioning
          sensitive: false
          value: "false"
        - hcl: false
          name: name
          sensitive: false
          value: mntl-test-workspace
        - hcl: false
          name: output_secret_name
          sensitive: false
          value: test-namespace/mntl-test-workspace/s3
        - hcl: true
          name: secret_tags
          sensitive: false
          value: '{access-project = "test-namespace-ops"}'
        - hcl: true
          name: tags
          sensitive: false
          value: |-
            {
              Application = "mntl-test-workspace"
              Component = "mntl-test-workspace"
              Owner = "sre"
              Project = "test-project"
            }
        - hcl: false
          name: tfcloud_agent
          sensitive: false
          value: "true"
      terraformVersion: 1.3.10
      token:
        secretKeyRef:
          key: token
          name: terraformrc
  2: |
    apiVersion: app.terraform.io/v1alpha2
    kind: Module
    metadata:
      annotations:
        app.mintel.com/altManifestFileSuffix: mntl-test-workspace-s3
        app.mintel.com/placeholder: placeholder
        app.mintel.com/terraform-allow-destroy: "true"
        app.mintel.com/terraform-cloud-tags: env:dev,owner:sre,mod:s3
        app.mintel.com/terraform-owner: sre
        argocd.argoproj.io/sync-options: SkipDryRunOnMissingResource=true
        argocd.argoproj.io/sync-wave: "-40"
      labels:
        app.kubernetes.io/name: mntl-test-workspace-s3
        app.mintel.com/env: dev
        app.mintel.com/owner: sre
        app.mintel.com/region: eu-west-1
        name: mntl-test-workspace-s3
      name: dev-eu-west-1-cluster1-test-namespace-mntl-test-workspace-s3
      namespace: test-namespace
    spec:
      module:
        source: app.terraform.io/Mintel/private-s3-bucket/aws
        version: 3.0.2
      terraformVersion: 1.3.10
      token:
        secretKeyRef:
          key: token
          name: terraformrc
      variables:
        - name: aws_account_name
        - name: aws_region
        - name: eks_cluster_name
        - name: enable_versioning
        - name: name
        - name: output_secret_name
        - name: secret_tags
        - name: tags
        - name: tfcloud_agent
      workspace:
        name: dev-eu-west-1-cluster1-test-namespace-mntl-test-workspace-s3
